routes/common/postgres_helper.rb

```diff
@@ -113,34 +113,24 @@ def delete_firewall_rule(firewall_rule_ubid)
   def restore
     Authorization.authorize(@user.id, "Postgres:create", project.id)
     Authorization.authorize(@user.id, "Postgres:view", @resource.id)
-    if @mode == AppMode::API
-      required_parameters = ["name", "restore_target"]
-      request_body_params = Validation.validate_request_body(@request.body.read, required_parameters)
 
-      st = Prog::Postgres::PostgresResourceNexus.assemble(
-        project_id: project.id,
-        location: @resource.location,
-        name: request_body_params["name"],
-        target_vm_size: @resource.target_vm_size,
-        target_storage_size_gib: @resource.target_storage_size_gib,
-        parent_id: @resource.id,
-        restore_target: request_body_params["restore_target"]
-      )
+    required_parameters = ["name", "restore_target"]
+    request_body_params = Validation.validate_request_body(params, required_parameters)
+
+    st = Prog::Postgres::PostgresResourceNexus.assemble(
+      project_id: project.id,
+      location: @resource.location,
+      name: request_body_params["name"],
+      target_vm_size: @resource.target_vm_size,
+      target_storage_size_gib: @resource.target_storage_size_gib,
+      parent_id: @resource.id,
+      restore_target: request_body_params["restore_target"]
+    )
 
+    if @mode == AppMode::API
```

Comment:
initializer now raises error on unexpected app class, and as you said we do lots of checks in the app code. I have some ideas to remove some of the branching, but I didn't want to do that now as it would make this patch even more complex. So... until then, how about keeping these as is?